models/DataObject/Fieldcollection/Dao.php

```diff
@@ -248,12 +248,20 @@ public function delete(DataObject\Concrete $object, bool $saveMode = false): arr
             }
         }
 
-        $where = "(ownertype = 'fieldcollection' AND " . Helper::quoteInto($this->db, 'ownername = ?', $this->model->getFieldname())
-            . ' AND ' . Helper::quoteInto($this->db, 'src_id = ?', $object->getId()) . ')';
-
-        // empty relation table
-        $this->db->executeStatement('DELETE FROM object_relations_' . $object->getClassId() . ' WHERE ' . $where);
-        $this->db->executeStatement('DELETE FROM object_relations_' . $object->getClassId() . ' WHERE ' . $whereLocalizedFields);
+        if (!$saveMode) {
+            $where = "(ownertype = 'fieldcollection' AND ".Helper::quoteInto(
+                    $this->db,
+                    'ownername = ?',
+                    $this->model->getFieldname()
+                )
+                .' AND '.Helper::quoteInto($this->db, 'src_id = ?', $object->getId()).')';
+
+            // empty relation table
+            $this->db->executeStatement('DELETE FROM object_relations_'.$object->getClassId().' WHERE '.$where);
+            $this->db->executeStatement(
+                'DELETE FROM object_relations_'.$object->getClassId().' WHERE '.$whereLocalizedFields
+            );
```

Comment:
```suggestion
            if ($hasLocalizedFields){
                  $this->db->executeStatement(
                      'DELETE FROM object_relations_'.$object->getClassId().' WHERE '.$whereLocalizedFields
                  );
            }
```
but i am not sure if is correct and why the former code looked like that (with some repetition), It probably needs to be refactored along 
https://github.com/pimcore/pimcore/pull/16482/files#diff-7011a9d08159e8735c4312cbc3c25f66574317e745300d8c913181bc55eb280cR245